### ISSUE TITLE
Luminance / Luma confusion

### ISSUE THREAD

**Issue by xi** [w3c/wcag#236]
Some time ago filed a [bug report with Less](https://github.com/less/less.js/issues/2829) because I thought they may have confused the terms "luminance" and "luma". They use luma to calculate the color contrast although the [spec](https://www.w3.org/TR/WCAG20/#contrast-ratiodef) clearly states that you have to use "relative luminance".

We came to the conclusion that Less is actually right and the spec may be wrong. From an [article by Charles Poynton](http://www.poynton.com/PDFs/YUV_and_luminance_harmful.pdf) I understand that the term "luminance" has been used ambiguously in the past and the term "luma" has been specifically invented to allow a clear distinction. The wrong wording in the spec may therefore be a relic of that ambiguous past.

So am I correct in thinking that the spec actually refers to luma? If that is the case, how can it be fixed? IMHO there should at least be a note about the luminance/luma ambiguity in the ["Understanding WCAG 2.0" document](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html).

_I have already brought this up on the mailinglist some time ago, without much success_

**Comment by moekraft** [comment]
hi

**Comment by moekraft** [comment]
Hi Xi, I am researching. Thanks, Moe

**Comment by moekraft** [comment]
Hi Xi, 

I have read through your materials and I do believe the WCAG definition of relative luminance needs clarification. The formula used in Notes 1 is correct, however, the conversion of the gamma-corrected R'G'B' value I believe needs some clarification. 

If you read through the formula it's proposing to covert the non-linear RGB values to linear RGB values, e.g. R'sRGB = 12.92 x RsRGB (gamma-corrected) and the formula is reversing that RsRGB = R'sRB/12.92 (linear). The condition checked R,G,B <=.03928 is coming straight from the sRGB spec on which this is based although this seems backwards to me since we are trying to solve for R,G,B. 

I disagree that the spec is trying to define luma while it is converting luma (R'G'B') to RGB by removing the gamma correctness. 

I will work on a proposed update to the definition to clarify. But need to pull in some experts to review. 

Thanks,
Moe

**Comment by xi** [comment]
@moekraft Thanks for working on this! What is the current status? (ie can we close this issue?)

As far as I understand #256, you concluded that the wording in the spec is right and the wording in LESS is wrong. Could you please verify that and maybe summarize your findings so that I can convince the LESS authors to change their wording?

**Comment by moekraft** [comment]
@xi Hi Tobias, I had proposed an update to the definition in the WCAG Glossary, however, since this is normative, I could not submit a pull request against the glossary. It was recommended that I update the notes in the understanding document where relevant. It's been a few months since I worked on this one. I held off since we were focused on WCAG 2.1 Success Criteria. Let me review the proposed change, identify where this clarification can be applied in the understanding document and summarize for you.

**Comment by peteroupc** [comment]
The value 0.03928 (as used in a note to the "relative luminance" definition) comes from the sRGB proposal.  The standard threshold in question is 0.04045.  (However, as far as I can tell, this difference doesn't affect the result where 8-bit-per-component sRGB colors are involved.)  Moreover, according to Poynton's article, "luma" is the proper term to use with respect to _nonlinearized_ RGB colors, rather than "luminance", which more properly refers to the perceived brightness given a _linearized_ RGB color.

**Comment by peteroupc** [comment]
Portion stricken out because I don't know whether 0.04045 or  12.92 * 0.0031308 = 0.040449936 is used in the IEC standard.

**Comment by danburzo** [comment]
Hi there, I was looking at the relative luminance computation for a library I'm working on. As far as I can [tell from this paper](https://www.w3.org/Graphics/Color/srgb), and [corroborated with Wikipedia](https://en.wikipedia.org/wiki/SRGB#The_sRGB_transfer_function_(%22gamma%22)), the value defined in the sRGB standard seems to be 0.04045.

**Comment by peteroupc** [comment]
The first link is to a paper which appears to come from the International Color Consortium and was uploaded to the W3C site at an unknown time (its URL ends with "srgb"; this is not to be confused with the sRGB proposal with a URL ending in "sRGB").  Does IEC 61966-2-1 really specify the inverse color component transfer function, or is that implied (not directly stated) in the standard?

**Comment by Myndex** [comment]
Hi @danburzo @peteroupc @moekraft @xi 

I see this issue is still open, and it is also discussed in issue #360, and is discussed more in depth over there (this issue should probably be merged??). It is also a tangential part of #695, which is looking into more advanced contrast math.

The threshold for sRGB to Y (Luminance) is <= 0.04045 for the linear portion, as DEFINED in the IEC standard, and can be seen in the freely available ITU document: https://www.itu.int/dms_pub/itu-r/opb/rep/R-REP-BT.2380-2-2018-PDF-E.pdf

Note however, that there are some gamma encoded transforms showing coefficients for NTSC (Rec601). There is to be sure a heap-ton of confusion on this subject.

I work in Hollywood in Feature Films and Television, specifically with digital imaging. 

The original sRGB draft had math that resulted in a discontinuity. This was addressed as pat of the IEC standard, though it's possible to be "more accurate" by not rounding as they did. But the rounded 0.04045 is perfectly suitable for the purpose of the WCAG. (There is a more significant problem is how contrast rations are chosen being discussed in issue #695).

### But also quickly:

**LUMINANCE**: this is a LINEAR (straight line) value based on light measurements. It is the **Y** from CIEXYZ, where it is a normalized value (0-100 or 0-1 for black to white). It is also **L** when measuring light, either reflective or emissive. Those are normally measures as cd/m^2 AKA "nits".

**LUMA**: This means the non-linear gamma encoded Y´(Y´prime) as used in video encoded like Y´IQ. sRGB does not use LUMA, unless it is the YCC version. The R´G´B´of sRGB are tristimulus components that each have been gamma encoded. 

__L* :__ aka **Lstar** — this is "perceptual lightness" as used in CIELAB (L* a* b*) it is non-linear and based specifically on human perception. CIELAB is intended to be perceptually uniform.

**sRGB**: the "proper" sRGB transfer curve has a linearized portion (the aforementioned threshold of 0.04045 for the gamma encoded R´G´B´). However, it is very common to use a simple 1/2.2 power curve to define the gamma. The purpose of the linearized section is to prevent the math problem of an infinite slope at 0. The "proper" curve is most useful when doing transforms of the data. The simple curve is what is used by old CRT monitors and analog circuitry among other things.

**Comment by xi** [comment]
I agree that this, #360, and #695 are related in that they all discuss color contrat. But they are not the same issue at all. This one is only which term should be used: "Luminance" or "Luma".

**Comment by danburzo** [comment]
@Myndex thank you for the clarifications and the links to related issues!

**Comment by Myndex** [comment]
> I agree that this, #360, and #695 are related in that they all discuss color contrat. But they are not the same issue at all. This one is only about which term should be used: "Luminance" or "Luma".

I see, I was looking at the many posts regarding the incorrect math in the WCAG.

THAT SAID, **the correct term is LUMINANCE.** The WCAG's current contrast assessment method uses luminance. 

LUMA is not the correct term, luma refers _only_ to y prime (Y´) as used in certain video encoding types.

**Comment by alastc** [comment]
Proposed response (taken form above):

Luminance is the correct term for WCAG's usage. LUMA is not the correct term, luma refers only to y prime (Y´) as used in certain video encoding types.
There are other updates in progress (e.g. #360), but they are separate from this issue.

**Comment by Myndex** [comment]
> Proposed response (taken from above):
> 
> Luminance is the correct term for WCAG's usage. LUMA is not the correct term, luma refers only to y prime (Y´) as used in certain video encoding types.
> There are other updates in progress (e.g. #360), but they are separate from this issue.

Yes — not sure where this is going to be referenced? if you wanted to add fo clarity: 

_LUMINANCE_ is a linear measure of light (in either relative or absolute terms), and _LUMA_ is encoded with a "gamma" curve, as such is non-linear.

**Comment by alastc** [comment]
Not that I'm wanting to do work for people running LESS, but I guess the answer would be better if it directly addressed the article by Charles Poynton referenced at the top. However, for our purpose I just want to be clear WCAG is using the right term (compared to luma). Thanks.

**Comment by alastc** [comment]
The [proposed response](https://github.com/w3c/wcag/issues/236#issuecomment-500375224) was accepted by the working group.
